### ISSUE TITLE
Apple Pay / Google Pay buttons do not show on product pages for a bookable product

### ISSUE THREAD

**Issue by thuautp** [Automattic/woocommerce-payments#4444]
### Describe the bug
The Apple Pay/ Google Pay buttons don't show on product pages when it's a booking product.

### To Reproduce
1. Install and activate WooCommerce Bookings.
2. Create a product with type Bookable product. Make sure with go to Costs > Base cost > fill in something so it’s not free.
3. Go to that product page.
4. See that Apple Pay/ Google Pay buttons not showing

### Expected behavior
Express Checkout should work with Booking products, according to our public doc: https://woocommerce.com/document/payments/apple-pay/#product-compatibility

### Devices

The merchant tested on the following devices with the latest versions:

1) iPhone 13s Max - Safari
2) iPad 7th Gen - Safari
3) Onn 8” Android v10 Tablet - Chrome
4) TCL Android A600DL phone with Android v11 - Chrome

### Additional context
The issue was found in 5366434-zen

There's a discussion with devs in Slack C7U3Y3VMY/p1657667813491869 where @shendy-a8c found that WCPay hooks into `woocommerce_after_add_to_cart_quantity` [here](https://github.com/Automattic/woocommerce-payments/blob/f4ef1de411128245f8cd617e088f4803fca8b297/includes/class-wc-payments-payment-request-button-handler.php#L74) but [WooCommerce Bookings’ template for bookable product](https://github.com/woocommerce/woocommerce-bookings/blob/7fc5db1788b74f4961b1efbd958a36d1a75313a2/templates/single-product/add-to-cart/booking.php), doesn’t trigger that hook. When they added `add_action( 'woocommerce_before_add_to_cart_button', [ $this, 'display_payment_request_button_html' ], 1 );` after this [line](https://github.com/Automattic/woocommerce-payments/blob/f4ef1de411128245f8cd617e088f4803fca8b297/includes/class-wc-payments-payment-request-button-handler.php#L74), the payment request button shows but got **Billing Street address is a required field.**

**Comment by mordeth** [comment]
Blocked by pdjVC1-12Q-p2

**Comment by frosso** [comment]
Reporting on the latest status for clarity:

> A year ago, we worked on this issue and I marked it as blocked because of the difficulty in incorporating bookable products into the cart when the Apple Pay button is triggered.

> We started pdjVC1-12Q-p2 with the Proton team regarding the correct method for adding bookable products to the cart through code. Despite attempting their suggested solution, it proved unsuccessful, and subsequently, I haven’t had the opportunity to delve deeper into debugging this problem.

**Comment by pierorocca** [comment]
@hsingyuc wasn't this resolved recently in another issue?

**Comment by hsingyuc** [comment]
@thuautp Were you testing with a bookable product that requires confirmation?  If the bookable product requires confirmation, the express checkout buttons will not be rendered because shoppers can only pay for the order after receiving a confirmation from the merchant.

**Comment by vbelolapotkov** [comment]
@mordeth I suppose you're not working on this one anymore, so removing you from assignee. Feel to self-assign again if I made a mistake.
